From 6ad43f64f8b7804c932e511d27f82de52846c27c Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Wed, 11 Jan 2017 14:42:02 +0800
Subject: [PATCH 1469/5242] MLK-13741: video: mxsfb: unchecked return value

commit  9cc10e5c2549822ff9f289d87a6d4802486a075d from
https://source.codeaurora.org/external/imx/linux-imx.git

In overlayfb_enable(), unchecked the return value of lock_fb_info function, if
it return zero, it maybe cause mistakes.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index fed7c7d..7fa2a09 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1557,7 +1557,9 @@ static void overlayfb_enable(struct mxsfb_layer *ofb)
 {
 	struct mxsfb_info *fbi = ofb->fbi;
 
-	lock_fb_info(fbi->fb_info);
+	if (!lock_fb_info(fbi->fb_info))
+		return;
+
 	if (fbi->cur_blank == FB_BLANK_UNBLANK) {
 		mxsfb_disable_controller(fbi->fb_info);
 		writel(CTRL1_FIFO_CLEAR, fbi->base + LCDC_CTRL1 + REG_SET);
-- 
1.7.9.5

